src/python/pants/testutil/rule_runner.py

```diff
@@ -142,9 +143,17 @@ def target_types(self) -> FrozenOrderedSet[Type[Target]]:
         return self.build_config.target_types
 
     def request(self, output_type: Type[_O], inputs: Iterable[Any]) -> _O:
-        result = assert_single_element(
-            self.scheduler.product_request(output_type, [Params(*inputs)])
-        )
+        # TODO: Update all callsites to pass this explicitly via session values.
+        session = self.scheduler
+        for value in inputs:
+            if type(value) == OptionsBootstrapper:
```

Comment:
should you then pop this value from the `inputs`?

Reply:
Yea, you could. It's unused now.